homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -62,16 +58,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
 
     if "amc:api" not in entry.data["token"]["scope"]:
-        ir.async_create_issue(
-            hass,
-            DOMAIN,
-            "wrong_scope",
-            is_fixable=True,
-            is_persistent=True,
-            severity=ir.IssueSeverity.WARNING,
-            translation_key="wrong_scope",
-            data={"entry_id": entry.entry_id},
-        )
+        entry.async_start_reauth(hass)
```

Comment:
We can raise `ConfigEntryAuthFailed`.

Reply:
Ok. Then please add a code comment here explaining why we don't raise.

Reply:
I'm reading the linked issue and it seems not having the websocket will create unexpected behavior and problems. Why should we allow to use the integration without the websocket?

Reply:
I thought it's better for the user. If the user updates its HA instance, the the integration is more or less broken, until he finds time to reauthenticate. And so he can at least use the integration with polling every 8 minutes. But if you think it's better to raise `ConfigEntryAuthFailed` we can do so.

Reply:
I think it's better to have things behave predictably. The docs say you should activate both APIs, so most users will probably have done that already.